RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc

```diff
@@ -463,8 +467,21 @@ int SiPixelTemplateReco::PixelTempReco1D(int id,
 
   // next, center the cluster on template center if necessary
 
-  midpix = (fypix + lypix) / 2;
-  shifty = templ.cytemp() - midpix;
+  if (goodEdgeAlgo) {
+    if (cotbeta >= 0) {
+      midpix = (int)(fypix + 0.5 * cotbeta * templ.zsize() / ysize);
+    } else {
+      midpix = (int)(lypix + 0.5 * cotbeta * templ.zsize() / ysize);
+    }
+  }  //if(goodEdgeAlgo)
+  else {
+    midpix = (fypix + lypix) / 2;
+    shifty = templ.cytemp() - midpix;
+  }  //else
+
+  shifty = BHY - midpix;
```

Comment:
This overwrites [L479](https://github.com/cms-sw/cmssw/pull/48008/files#diff-4d7cf7408dd2325ed83d990527638d2b82bf22e17b442d50d4a8352b1b73a413R479)